l10n_fr_intrastat_service/models/intrastat_service.py

```diff
@@ -0,0 +1,378 @@
+        (3, '03'),
+        (4, '04'),
+        (5, '05'),
+        (6, '06'),
+        (7, '07'),
+        (8, '08'),
+        (9, '09'),
+        (10, '10'),
+        (11, '11'),
+        (12, '12')
+        ], string='Month', required=True,
+        default=lambda self: self._get_month(),
+        states={'done': [('readonly', True)]})
+    year_month = fields.Char(
+        compute='_compute_year_month', string='Period', readonly=True,
+        track_visibility='onchange', store=True,
+        help="Year and month of the declaration.")
+    declaration_line_ids = fields.One2many(
+        'l10n.fr.intrastat.service.declaration.line',
+        'parent_id', string='DES Lines',
+        states={'done': [('readonly', True)]}, copy=False)
+    num_decl_lines = fields.Integer(
+        compute='_compute_numbers', string='Number of Lines', store=True,
+        readonly=True, track_visibility='always')
+    total_amount = fields.Monetary(
+        compute='_compute_numbers', currency_field='currency_id',
+        string='Total Amount', store=True, track_visibility='onchange',
+        readonly=True,
+        help="Total amount in company currency of the declaration.")
+    currency_id = fields.Many2one(
+        related='company_id.currency_id', string='Company Currency',
+        store=True)
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('done', 'Done'),
+        ], string='State', readonly=True, track_visibility='onchange',
+        default='draft', copy=False)
+
+    _sql_constraints = [(
+        'date_uniq', 'unique(year_month, company_id)',
+        'A DES already exists for this month!'
+        )]
+
+    def _prepare_domain(self):
+        start_date = date(self.year, self.month, 1)
+        end_date = start_date + relativedelta(day=1, months=1, days=-1)
+        domain = [
+            ('type', 'in', ('out_invoice', 'out_refund')),
+            ('date_invoice', '<=', end_date),
+            ('date_invoice', '>=', start_date),
+            ('state', 'in', ('open', 'in_payment', 'paid')),
+            ('company_id', '=', self.company_id.id),
+            ]
+        return domain
+
+    def _is_service(self, invoice_line):
+        if invoice_line.product_id.type == 'service':
+            return True
+        else:
+            return False
+
+    def generate_service_lines(self):
+        self.ensure_one()
+        line_obj = self.env['l10n.fr.intrastat.service.declaration.line']
+        invoice_obj = self.env['account.invoice']
+        eur_cur = self.env.ref('base.EUR')
+        self._check_generate_lines()
+        # delete all DES lines generated from invoices
+        lines_to_remove = line_obj.search([
+            ('invoice_id', '!=', False),
+            ('parent_id', '=', self.id)])
+        lines_to_remove.unlink()
+
+        invoices = invoice_obj.search(
+            self._prepare_domain(), order='date_invoice')
+        for invoice in invoices:
+            if not invoice.partner_id.country_id:
```

Comment:
Shouldn't it be based on `commercial_partner_id`?

Reply:
ping @alexis-via

Reply:
yes, I'm not against changing that...
But I would really like to understand the problem with message_post()... it seems to me it is a different between python 2 and 3, but I'm still not sure about it.